### PR TITLE
[IMP] web_editor improve view on mobile devices

### PR DESCRIPTION
**Pull request by res-odoo**
Tasks : [1929027](https://www.odoo.com/web#id=1929027&action=333&active_id=1520&model=project.task&view_type=form&menu_id=4720)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by adr-odoo** [comment]
Waiting merge of new version of web_editor...

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        this._super.apply(this, arguments);
+        // Create a reference for bind function
+        this._computeScrollBarIconPosition = _.throttle(this._computeScrollBarIconPosition.bind(this), 100);
+        this._hideDropdownMenuShow = _.throttle(this._hideDropdownMenuShow.bind(this), 100);
+    },
+    /**
+     * Helper for query the $content node using vanilla js
+     *
+     * @private
+     * @param cssSelector
```
Comment (Gorash) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: Add the kind of variable

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+}
+var core = require('web.core');
+var _t = core._t;
+
+var FieldHtml = require('web_editor.field.html');
+FieldHtml.include({
+    /**
+     * Append throttle function for event append inside the _onLoadWysiwyg
+     * @override
+     * @param parent
```
Comment (Gorash) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: Add the kind of variable

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+     * @returns {HTMLElement}
+     */
+    _queryContentNode: function(cssSelector) {
+        return this.$content ? this.$content.get(0).querySelector(cssSelector) : null;
+    },
+    /**
+     * Determine if the note-toolbar node needed to show the scrollable-start and scrollable-end
+     *
+     * @private
+     * @param event
```
Comment (Gorash) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: Add the kind of variable + all next

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        if (event) {
+            event.preventDefault();
+        }
+        var toolbarNode = this._queryContentNode('.note-toolbar');
+        var toolbarWrapperNode = this._queryContentNode('.note-toolbar-wrapper');
+        if (toolbarNode == null || toolbarWrapperNode == null) {
+            return;
+        }
+
+        var computeToolBarWrapperScrollableClass = function (className, canRemoveClass) {
```
Comment (Gorash) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: There is no way to put this in private method on the object?

**Comment by pparidans** [comment]
Why is your branch in `master`but your PR against `saas-12.2` ? :thinking:

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+FieldHtml.include({
+    /**
+     * Append throttle function for event append inside the _onLoadWysiwyg
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        // Create a reference for bind function
+        this._computeScrollBarIconPosition = _.throttle(this._computeScrollBarIconPosition.bind(this), 100);
+        this._hideDropdownMenuShow = _.throttle(this._hideDropdownMenuShow.bind(this), 100);
```
Comment (adr-odoo) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: Should be prefixed by _throttleHide...

File: addons/web_editor/static/src/js/backend/field_html_mobile.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+     *
+     * @override
+     */
+    _onLoadWysiwyg: function () {
+        this._super.apply(this, arguments);
+        document.addEventListener('scroll', this._hideDropdownMenuShow.bind(this));
+        this._appendCancelNodeToDropdown();
+        // use a setTimeout because we need to wait the css computing and rendering for knowing
+        // if an scrollbar is displayed on the screen
+        setTimeout(this._appendScrollIcon.bind(this));
```
Comment (adr-odoo) [inline addons/web_editor/static/src/js/backend/field_html_mobile.js]: ? We don't know how much time to wait...

Comment (adr-odoo) [reply addons/web_editor/static/src/js/backend/field_html_mobile.js]: Or add a fixme with future improvements to do

**Comment by res-odoo** [comment]
Moved to https://github.com/odoo/odoo/pull/33110

**Pull request closed without merging**